From 4bd9f4f9c86776cb62a1ff10bfeec97ae4593e14 Mon Sep 17 00:00:00 2001
From: Xulin Sun <xulin.sun@windriver.com>
Date: Mon, 13 Jun 2016 14:20:22 +0800
Subject: [PATCH 8/8] dpio/qbman_portal: simplify logical expression

fsl-mc/bus/dpio/qbman_portal.c:242:22: warning: logical not is only applied to
the left hand side of comparison [-Wlogical-not-parentheses]
  BUG_ON(!p->mc.check != swp_mc_can_submit);
                        ^

Signed-off-by: Xulin Sun <xulin.sun@windriver.com>
---
 drivers/staging/fsl-mc/bus/dpio/qbman_portal.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/drivers/staging/fsl-mc/bus/dpio/qbman_portal.c b/drivers/staging/fsl-mc/bus/dpio/qbman_portal.c
index 6c5638b..d859ddb 100644
--- a/drivers/staging/fsl-mc/bus/dpio/qbman_portal.c
+++ b/drivers/staging/fsl-mc/bus/dpio/qbman_portal.c
@@ -239,7 +239,7 @@ void qbman_swp_mc_submit(struct qbman_swp *p, void *cmd, uint32_t cmd_verb)
 {
 	uint32_t *v = cmd;
 #ifdef QBMAN_CHECKING
-	BUG_ON(!p->mc.check != swp_mc_can_submit);
+	BUG_ON(p->mc.check == swp_mc_can_submit);
 #endif
 	/* TBD: "|=" is going to hurt performance. Need to move as many fields
 	 * out of word zero, and for those that remain, the "OR" needs to occur
-- 
1.7.5.4

